SWLOR.Game.Server/Feature/MigrationDefinition/PlayerMigration/_8_UpgradeWeapons.cs

```diff
@@ -0,0 +1,160 @@
+            UpdateWeapons(player);
+        }
+
+        public Dictionary<string, (int, int)> itemReplace = new()
+            {
+                { "tit_rifle", (12, 15) },
+                { "cap_rifle", (15, 20) },
+                { "mando_rifle", (13, 17) },
+                { "h_rifle_2", (14, 24) },
+                { "del_rifle", (17, 27) },
+                { "poach_rifle", (18, 29) },
+                { "byysk_rifle", (18, 29) },
+                { "h_rifle_3", (19, 32) },
+                { "proto_rifle", (21, 34) },
+                { "raider_rifle", (22, 36) },
+                { "h_rifle_4", (23, 38) },
+                { "oph_rifle", (26, 39) },
+                { "h_rifle_5", (28, 42) },
+                { "b_twinblade", (6, 8) },
+                { "trn_saberstaff_1", (6, 8) },
+                { "twin_elec_1", (6, 8) },
+                { "tit_twinblade", (10, 13) },
+                { "trn_saberstaff_2", (10, 13) },
+                { "twin_elec_2", (10, 13) },
+                { "sith_twinblade", (11, 15) },
+                { "cap_twinblade", (13, 15) },
+                { "cap_saberstaff", (13, 15) },
+                { "mando_twinblade", (11, 13) },
+                { "mando_sabstaff", (11, 13) },
+                { "h_twinblade_2", (11, 16) },
+                { "h_twinelec_2", (11, 16) },
+                { "del_twinblade", (15, 18) },
+                { "trn_saberstaff_3", (15, 18) },
+                { "twin_elec_3", (15, 18) },
+                { "byysk_twinblade", (17, 19) },
+                { "poach_twinblade", (16, 19) },
+                { "h_twinblade_3", (16, 20) },
+                { "h_twinelec_3", (16, 20) },
+                { "proto_twinblade", (19, 22) },
+                { "trn_saberstaff_4", (19, 22) },
+                { "twin_elec_4", (19, 22) },
+                { "raider_twinblade", (20, 24) },
+                { "h_twinblade_4", (20, 25) },
+                { "h_twinelec_4", (20, 25) },
+                { "oph_twinblade", (24, 27) },
+                { "trn_saberstaff_5", (24, 27) },
+                { "twin_elec_5", (24, 27) },
+                { "h_twinblade_5", (25, 28) },
+                { "h_twinelec_5", (25, 28) }
+            };
+
+        private static void RefundPerks(Player dbPlayer, uint player)
+        {
+            List<PerkType> refundList = new(3)
+            {
+                PerkType.DualWield,
+                PerkType.ImprovedTwoWeaponFightingOneHanded,
+                PerkType.ImprovedTwoWeaponFightingTwoHanded
+            };
+
+            foreach (var toRefund in refundList)
+            {
+                if (!dbPlayer.Perks.ContainsKey(toRefund))
+                    continue;
+
+                dbPlayer.UnallocatedSP += 2;
+            }
+
+            if(dbPlayer.Perks.ContainsKey(PerkType.RapidShot))
+            {
+                var perkDetail = Perk.GetPerkDetails(PerkType.RapidShot);
+                var refundAmount = perkDetail.PerkLevels[0].Price;
```

Comment:
The price needs to be a hard coded value. The reason for this is because future updates to the price of this perk will throw off any migrations performed. It will work for now, but will break in the future.